include/siminusminus/containers/inmutableiterator.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef SIMINUSMINUS_CONTAINERS_INMUTABLEITERATOR_HPP
+#define SIMINUSMINUS_CONTAINERS_INMUTABLEITERATOR_HPP
+
+#include <algorithm>
+
+namespace cmm {
+namespace containers {
+
+/**
+ * \ingroup containers
+ * \brief iterator to inmutable objects.
+ * 
+ * The InmutableIterator class allows to create an iterator which cannot be
+ * modified, only allows read operations.
+ */
+template <typename T>
+class InmutableIterator : public std::iterator<std::input_iterator_tag, // iterator_category
```

Comment:
The convention is to name const_iterator iterators with no write "rights", and iterator to iterator classes with write "rights". Since this template is not specific to a container but a generic iterator for contiguous data structures, I would name it ConstContiguousIterator or something like that.